### PR TITLE
Rdcc 195: Create an entry into user_account_map table with pbaccount info.

### PR DESCRIPTION
**Pull request by KOTLAPRA23**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-195

### Change description ###

Create an entry into user_account_map table with pbaccount associated with an organisation.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/PaymentAccountServiceTest.java [hunk 1]
```diff
@@ -9,8 +9,15 @@
 import org.junit.Test;
 
 import org.powermock.api.mockito.PowerMockito;
-import uk.gov.hmcts.reform.professionalapi.domain.*;
```
Comment (akiohmcts) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/PaymentAccountServiceTest.java]: Please remove this in your IDEA

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java [hunk 2]
```diff
@@ -150,6 +159,18 @@ private void addDxAddressToContactInformation(List<DxAddressCreationRequest> dxA
         }
     }
 
+    private void persistedUserAccountMap(ProfessionalUser persistedSuperUser, List<PaymentAccount> paymentAccounts) {
+
+        if (paymentAccounts != null
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java]: CollectionUtils.isEmpty method can be used

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java [hunk 3]
```diff
@@ -150,6 +159,18 @@ private void addDxAddressToContactInformation(List<DxAddressCreationRequest> dxA
         }
     }
 
+    private void persistedUserAccountMap(ProfessionalUser persistedSuperUser, List<PaymentAccount> paymentAccounts) {
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java]: method name can be more meaningfull like persistUserAccountMap

File: src/main/java/uk/gov/hmcts/reform/professionalapi/domain/UserAccountMapId.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Embeddable
+@Getter
+@NoArgsConstructor
+public class UserAccountMapId implements Serializable {
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/domain/UserAccountMapId.java]: is Serializable is required ?

File: build.gradle [hunk 5]
```diff
@@ -173,7 +173,7 @@ pitest {
     outputFormats = ['XML', 'HTML']
     timestampedReports = false
     mutationThreshold = 90
-   
+
```
Comment (shreedhar-hmcts) [inline build.gradle]: unwanted line can be removed

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/PaymentAccountServiceTest.java (hunk 1 above)
Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/PaymentAccountServiceTest.java]: Its removed in the IDE

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java (hunk 2 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java]: in the next JIRA i will implement the above change

File: src/main/java/uk/gov/hmcts/reform/professionalapi/domain/UserAccountMapId.java (hunk 4 above)
Comment (akiohmcts) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/domain/UserAccountMapId.java]: I think we should leave it there

**Pull request merged**